screenshot-iframe.js

```diff
@@ -0,0 +1,80 @@
+import {storageHost} from './constants';
+import url from 'url';
+
+function parseQuery(queryString) {
+    const query = {};
+    const pairs = (queryString[0] === '?' ? queryString.substr(1) : queryString).split('&');
+    for (let i = 0; i < pairs.length; i++) {
+      const pair = pairs[i].split('=');
+      const k = decodeURIComponent(pair[0]);
+      if (k) {
+        const v = decodeURIComponent(pair[1] || '');
+        query[k] = v;
+      }
+    }
+    return query;
+}
+
+export const getScreenshotBlob = async (reqUrl, hash, ext, type, width, height) => {
+
+    var iframeExists = document.getElementsByName('screenshotIframe');
+
+    if(!iframeExists.length) {
+        var iframe = document.createElement('iframe');
+        iframe.width = '0px';
+        iframe.height = '0px';
+        iframe.name = 'screenshotIframe';
+        document.body.appendChild(iframe);
+    }
+
+    if(hash) {
+        var tempUrl = `https://app.webaverse.com/${hash}.${ext}/preview.${type}`
+    }
+    else {
+        var tempUrl = `https://app.webaverse.com/screenshot.html?url=${reqUrl}&hash=${hash}&ext=${ext}&type=${type}&width=${width}&height=${height}`;
+    }
+
+    const u = url.parse(tempUrl, true);
+
+    const spec = (() => {
+        if(!u.search){
+          const match = u.pathname.match(/^\/([^\.]+)\.([^\/]+)\/([^\.]+)\.(.+)$/);
+          if (match) {
+            let hash = match[1];
+            let ext = match[2].toLowerCase();
+            let type = match[4].toLowerCase();
+            let url = `${storageHost}/ipfs/${hash}`;
+            return {
+              url,
+              hash,
+              ext,
+              type
+            }
+          }
+        }
+        else if(u.search){
+            return parseQuery(u.search);
+        }
+        spec = null;
+    })();
+
+    if (spec) {
+        const {url, hash, ext, type, height, width} = spec;
+        var ssUrl = `https://app.webaverse.com/screenshot.html?url=${url}&ext=${ext}&type=${type}&width=${width}&height=${height}`;
+    }
+
+    window.open(ssUrl, 'screenshotIframe')
```

Comment:
Have you tested it that it's not re-creating the iframe I don't think so it's tested properly?

Reply:
Yes, I have tested it by multiple function calls and it's not creating new iframes for every preview.